test/Sentry.Tests/Internals/Http/CachingTransportTests.cs

```diff
@@ -0,0 +1,227 @@
+
+            using var transport = new CachingTransport(innerTransport, options);
+
+            // Act & assert
+            for (var i = 0; i < 20; i++)
+            {
+                using var envelope = Envelope.FromEvent(new SentryEvent());
+                await transport.SendEnvelopeAsync(envelope);
+
+                transport.GetCacheLength().Should().BeLessOrEqualTo(options.MaxQueueItems);
+            }
+        }
+
+        [Fact(Timeout = 10000)]
+        public async Task AwareOfExistingFiles()
+        {
+            // Arrange
+            var options = new SentryOptions {CacheDirectoryPath = CacheDirectoryPath};
+
+            // Send some envelopes with a failing transport to make sure they all stay in cache
+            {
+                var initialInnerTransport = new FakeFailingTransport();
+                using var initialTransport = new CachingTransport(initialInnerTransport, options);
+
+                for (var i = 0; i < 3; i++)
+                {
+                    using var envelope = Envelope.FromEvent(new SentryEvent());
+                    await initialTransport.SendEnvelopeAsync(envelope);
+                }
+            }
+
+            using var innerTransport = new FakeTransport();
+            using var transport = new CachingTransport(innerTransport, options);
+
+            // Act
+
+            // Wait until directory is empty
+            while (
+                Directory.Exists(CacheDirectoryPath) &&
+                Directory.EnumerateFiles(CacheDirectoryPath, "*", SearchOption.AllDirectories).Any())
+            {
+                await Task.Delay(100);
+            }
+
+            // Assert
+            innerTransport.GetSentEnvelopes().Should().HaveCount(3);
+        }
+
+        [Fact(Timeout = 10000)]
+        public async Task RetriesOnTransientExceptions()
+        {
+            // Arrange
+            var options = new SentryOptions {CacheDirectoryPath = CacheDirectoryPath};
+
+            var innerTransport = Substitute.For<ITransport>();
+            var isFailing = true;
+
+            innerTransport
+                .SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>())
+                .Returns(_ =>
+                    isFailing
+                        ? new ValueTask(Task.FromException(new IOException()))
+                        : new ValueTask()
+                );
+
+            using var transport = new CachingTransport(innerTransport, options);
+
+            // Can't really reliably test this with a worker
+            await transport.StopWorkerAsync();
+
+            // Act
+            for (var i = 0; i < 3; i++)
+            {
+                using var envelope = Envelope.FromEvent(new SentryEvent());
+                await transport.SendEnvelopeAsync(envelope);
+            }
+
+            await transport.FlushAsync();
+
+            // (transport stops failing)
+            innerTransport.ClearReceivedCalls();
+            isFailing = false;
```

Comment:
Can u make sure it didn't do the job yet?

```suggestion
            _ = innerTransport.Received(0).SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>());
            isFailing = false;
```

Reply:
well it did call that method when trying (and failing) initially